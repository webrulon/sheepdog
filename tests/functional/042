#!/bin/bash

# Test disk full handling

. ./common

_need_to_be_root

_make_device 0 $((1024 ** 3))
_make_device 1 $((1024 ** 3))
_make_device 2 $((30 * 1024 ** 2))
_make_device 3 $((30 * 1024 ** 2))

# create log files of sheep 2 and 3 on other places
touch $STORE/0/sheep2.log $STORE/0/sheep3.log
ln -s $STORE/0/sheep2.log $STORE/2/sheep.log
ln -s $STORE/0/sheep3.log $STORE/3/sheep.log

for i in `seq 0 3`; do
    _start_sheep $i
done

_wait_for_sheep 4

_cluster_format

# create two VDIs before there are enough spaces
$DOG vdi create test0 100M
$DOG vdi create test1 100M

# make sheep 0 and 1 disk full
dd if=/dev/zero of=$STORE/2/zero > /dev/null 2>&1
dd if=/dev/zero of=$STORE/3/zero > /dev/null 2>&1

# test data write against disk-full cluster
for i in `seq 0 10`; do
    echo $i | $DOG vdi write test0 $((i * 4 * 1024 * 1024)) 512 -p 7000
    echo $i | $DOG vdi write test1 $((i * 4 * 1024 * 1024)) 512 -p 7002
done

# test vdi creation against disk-full cluster
$DOG vdi create test2 100M -p 7000
$DOG vdi create test3 100M -p 7002

for i in `seq 0 3`; do
    $DOG node list -p 700$i
done
_node_info
ls $STORE/*/obj/* | _filter_store | sort
