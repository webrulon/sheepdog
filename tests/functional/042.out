QA output created by 042
using backend plain store
Failed to write object fd34af00000000: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000000: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000001: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000001: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000002: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000002: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000003: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000003: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000004: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000004: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000005: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000005: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000006: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000006: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000007: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000007: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000008: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000008: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af00000009: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc00000009: Server has no space for new objects
Failed to write VDI
Failed to write object fd34af0000000a: Server has no space for new objects
Failed to write VDI
Failed to write object fd32fc0000000a: Server has no space for new objects
Failed to write VDI
Failed to create VDI test2: Failed to write to requested VDI
Failed to create VDI test3: Failed to write to requested VDI
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	249          0
   1   127.0.0.1:7001      	249          1
   2   127.0.0.1:7002      	 7          2
   3   127.0.0.1:7003      	 7          3
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	249          0
   1   127.0.0.1:7001      	249          1
   2   127.0.0.1:7002      	 7          2
   3   127.0.0.1:7003      	 7          3
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	249          0
   1   127.0.0.1:7001      	249          1
   2   127.0.0.1:7002      	 7          2
   3   127.0.0.1:7003      	 7          3
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	249          0
   1   127.0.0.1:7001      	249          1
   2   127.0.0.1:7002      	 7          2
   3   127.0.0.1:7003      	 7          3
STORE	DATA	VDI	VMSTATE	ATTR	LEDGER	STALE
0	22	4	0	0	0	0
1	22	4	0	0	0	0
2	0	1	0	0	0	0
3	0	1	0	0	0	0
STORE/0/obj/00fd32fc00000000
STORE/0/obj/00fd32fc00000001
STORE/0/obj/00fd32fc00000002
STORE/0/obj/00fd32fc00000003
STORE/0/obj/00fd32fc00000004
STORE/0/obj/00fd32fc00000005
STORE/0/obj/00fd32fc00000006
STORE/0/obj/00fd32fc00000007
STORE/0/obj/00fd32fc00000008
STORE/0/obj/00fd32fc00000009
STORE/0/obj/00fd32fc0000000a
STORE/0/obj/00fd34af00000000
STORE/0/obj/00fd34af00000001
STORE/0/obj/00fd34af00000002
STORE/0/obj/00fd34af00000003
STORE/0/obj/00fd34af00000004
STORE/0/obj/00fd34af00000005
STORE/0/obj/00fd34af00000006
STORE/0/obj/00fd34af00000007
STORE/0/obj/00fd34af00000008
STORE/0/obj/00fd34af00000009
STORE/0/obj/00fd34af0000000a
STORE/0/obj/80fd32fc00000000
STORE/0/obj/80fd34af00000000
STORE/0/obj/80fd366200000000
STORE/0/obj/80fd381500000000
STORE/1/obj/00fd32fc00000000
STORE/1/obj/00fd32fc00000001
STORE/1/obj/00fd32fc00000002
STORE/1/obj/00fd32fc00000003
STORE/1/obj/00fd32fc00000004
STORE/1/obj/00fd32fc00000005
STORE/1/obj/00fd32fc00000006
STORE/1/obj/00fd32fc00000007
STORE/1/obj/00fd32fc00000008
STORE/1/obj/00fd32fc00000009
STORE/1/obj/00fd32fc0000000a
STORE/1/obj/00fd34af00000000
STORE/1/obj/00fd34af00000001
STORE/1/obj/00fd34af00000002
STORE/1/obj/00fd34af00000003
STORE/1/obj/00fd34af00000004
STORE/1/obj/00fd34af00000005
STORE/1/obj/00fd34af00000006
STORE/1/obj/00fd34af00000007
STORE/1/obj/00fd34af00000008
STORE/1/obj/00fd34af00000009
STORE/1/obj/00fd34af0000000a
STORE/1/obj/80fd32fc00000000
STORE/1/obj/80fd34af00000000
STORE/1/obj/80fd366200000000
STORE/1/obj/80fd381500000000
STORE/2/obj/80fd32fc00000000
STORE/3/obj/80fd34af00000000
